### ISSUE TITLE
clang-cl.exe: False positive C++17 warning in C++14: out-of-line definition of constexpr static data member

### ISSUE THREAD

**Issue by dmitrykobets-msft** [llvm/llvm-project#65689]
When compiling with C++14, I'm getting a warning that pertains only to C++17.
I tried this out with LLVM 17.0.0-rc4.

## Repro

```cpp
class A
{
    static constexpr const int i{0};
};
constexpr const int A::i;

int main()
{
}
```

`clang-cl.exe <thefile.cpp> /std:c++14 -Weverything`

**Output**
warning: out-of-line definition of constexpr static data member is redundant in C++17 and is deprecated [-Wdeprecated-redundant-constexpr-static-def]
